// ngx-i18next/build/browser/LanguageDetectorAdapter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = require("ts-smart-logger/index");
var LanguageDetectorAdapter = (function () {
    function LanguageDetectorAdapter() {
    }
    LanguageDetectorAdapter.toBrowserLanguageDetector = function (browserLanguageDetector) {
        var LanguageDetector = (function () {
            function LanguageDetector() {
                LanguageDetectorAdapter.logger.debug('[$LanguageDetectorAdapter][constructor] LanguageDetector wrapper has been instantiated');
            }
            LanguageDetector.prototype.init = function () {
            };
            LanguageDetector.prototype.detect = function () {
                var result = browserLanguageDetector.detect();
                LanguageDetectorAdapter.logger.debug('[$LanguageDetectorAdapter][detect] Detect method returns the language as', result);
                return result;
            };
            LanguageDetector.prototype.cacheUserLanguage = function () {
            };
            return LanguageDetector;
        }());
        LanguageDetector['type'] = 'languageDetector';
        return LanguageDetector;
    };
    return LanguageDetectorAdapter;
}());
LanguageDetectorAdapter.logger = index_1.LoggerFactory.makeLogger(LanguageDetectorAdapter);
exports.LanguageDetectorAdapter = LanguageDetectorAdapter;
